src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py

```diff
@@ -101,7 +101,7 @@ def _get_client(client, kwargs):
         directory_path = directory_path.replace('./', '', 1)
     dir_client = client.get_directory_client(directory_path=directory_path)
     file_name = kwargs.pop("file_name", None)
-    if file_name:
+    if file_name and file_name != '*':
```

Comment:
Is it possible that `directory_path` also contains `*`? Have we tested such case?